lib/twitter_cldr.rb

```diff
@@ -112,9 +112,24 @@ def locale_fallbacks
     end
 
     def convert_locale(locale)
```

Comment:
How about something like this? I was able to get (almost) all the tests to pass, and it maintains 100% compatibility:

```ruby
def convert_locale(locale)
  locale = normalize_locale(locale)

  unless supported_locale?(locale)
    loc = TwitterCldr::Shared::Locale.parse(locale)
    max_supported = loc.max_supported

    if loc.dasherized == 'und' || !max_supported
      return locale.to_sym
    end

    locale = normalize_locale(max_supported.dasherized)
  end

  locale.to_sym
end
```